gammapy/data/io.py

```diff
@@ -0,0 +1,112 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from typing import Optional
+from astropy.io import fits
+from astropy.table import Table
+from pydantic import BaseModel, ValidationError, validator
+from gammapy.utils.scripts import make_path
+from gammapy.utils.table_validator import GADF_EVENT_TABLE_DEFINITION, TableValidator
+
+
+class GADFEventsHeader(BaseModel):
+    OBS_ID: int
+    TSTART: float
+    TSTOP: float
+    ONTIME: float
+    LIVETIME: float
+    DEADC: float
+    OBS_MODE: str
+    RA_PNT: float
+    DEC_PNT: float
+    ALT_PNT: float
+    AZ_PNT: float
+    EQUINOX: float
+    RADECSYS: str
+    ORIGIN: str
+    TELESCOP: str
+    INSTRUME: str
+    CREATOR: str
+    HDUCLASS: str = "GADF"
+    HDUDOC: str = "https://gamma-astro-data-formats.readthedocs.io/en/v0.3/index.html"
+    HDUVERS: str = "v0.3"
+    HDUCLAS1: str = "EVENTS"
+
+    OBSERVER: Optional[str]
+    CREATED: Optional[str]
+    OBJECT: Optional[str]
+    RA_OBJ: Optional[float]
+    DEC_OBJ: Optional[float]
+    EV_CLASS: Optional[str]
+    TELAPSE: Optional[float]
+    TELLIST: Optional[str]
+    N_TELS: Optional[int]
+    TASSIGN: Optional[str]
+    DST_VER: Optional[str]
+    ANA_VER: Optional[str]
+    CAL_VER: Optional[str]
+    CONV_DEP: Optional[float]
+    CONV_RA: Optional[float]
+    CONV_DEC: Optional[float]
+    TRGRATE: Optional[float]
+    ZTRGRATE: Optional[float]
+    MUONEFF: Optional[float]
+    BROKPIX: Optional[float]
+    AIRTEMP: Optional[float]
+    PRESSURE: Optional[float]
+    RELHUM: Optional[float]
+    NSBLEVEL: Optional[float]
```

Comment:
Maybe, this long list could be into a file, the filename having also the version of the format... If you add a version id in the constructor of this class, you can have a generic validator of a given format and for all its versions.